syllabus/core/stat_recorder.py

```diff
@@ -34,45 +31,96 @@ def record(self, episode_return: float, episode_length: int, episode_task, env_i
+                    self.stats[episode_task]['mean_r_squared'] = round((self.stats[episode_task]['mean_r_squared'] * N_past + episode_return ** 2) / (N_past + 1), 4)
+                    self.stats[episode_task]['var_r'] = round(self.stats[episode_task]['mean_r_squared'] - self.stats[episode_task]['mean_r'] ** 2, 4)
+                    
+                    self.stats[episode_task]['mean_l'] = round((self.stats[episode_task]['mean_l'] * N_past + episode_length) / (N_past + 1), 4)
+                    self.stats[episode_task]['mean_l_squared'] = round((self.stats[episode_task]['mean_l_squared'] * N_past + episode_length ** 2) / (N_past + 1), 4)
+                    self.stats[episode_task]['var_l'] = round(self.stats[episode_task]['mean_l_squared'] - self.stats[episode_task]['mean_l'] ** 2, 4)
         else:
             raise ValueError("Unknown task")
-        
+    
+    def log_metrics(self, writer, step=None, log_full_dist=False):
+        """Log the statistics of the first 5 tasks to the provided tensorboard writer.
+
+        :param writer: Tensorboard summary writer.
         """
-        records = json.dumps(self.records, default=convert_numpy)
-        with open(os.path.join(self.write_path, 'records.json'), "w") as file:
-            file.write(records)
-        stats = json.dumps(self.stats, default=convert_numpy)
-        with open(os.path.join(self.write_path, 'stats.json'), "w") as file:
-            file.write(stats)
+        try:
+            import wandb
+            tasks_to_log = self.tasks
+            if len(self.tasks) > 5 and not log_full_dist:
+                warnings.warn("Only logging stats for 5 tasks.")
+                tasks_to_log = self.tasks[:5]
+            if self.task_names:
+                for idx in tasks_to_log:
+                    if self.stats[idx]:
+                        writer.add_scalar(f"stats_per_task/task_{self.task_space.task_name(idx)}_episode_return_mean", self.stats[idx]['mean_r'], step)
+                        writer.add_scalar(f"stats_per_task/task_{self.task_space.task_name(idx)}_episode_return_var", self.stats[idx]['var_r'], step)
+                        writer.add_scalar(f"stats_per_task/task_{self.task_space.task_name(idx)}_episode_length_mean", self.stats[idx]['mean_l'], step)
+                        writer.add_scalar(f"stats_per_task/task_{self.task_space.task_name(idx)}_episode_length_var", self.stats[idx]['var_l'], step)
+                    else:
+                        writer.add_scalar(f"stats_per_task/task_{self.task_space.task_name(idx)}_episode_return_mean", 0, step)
+                        writer.add_scalar(f"stats_per_task/task_{self.task_space.task_name(idx)}_episode_return_var", 0, step)
+                        writer.add_scalar(f"stats_per_task/task_{self.task_space.task_name(idx)}_episode_length_mean", 0, step)
+                        writer.add_scalar(f"stats_per_task/task_{self.task_space.task_name(idx)}_episode_length_var", 0, step)
```

Comment:
simplify this if else, theres a lot of repeated code. Also why are we logging 0 if there are no stats? We can probably just skip logging in that case